src/sqlparser/tests/testdata/qualified_operator.yaml

```diff
@@ -0,0 +1,28 @@
+# This file is automatically generated. See `src/sqlparser/test_runner/src/bin/apply.rs` for more information.
+- input: select 2 * 3 operator(pg_catalog.+) 4;
+  formatted_sql: SELECT 2 * 3 OPERATOR(pg_catalog.+) 4
+  formatted_ast: 'Query(Query { with: None, body: Select(Select { distinct: All, projection: [UnnamedExpr(BinaryOp { left: BinaryOp { left: Value(Number("2")), op: Multiply, right: Value(Number("3")) }, op: PGQualified(QualifiedOperator { schema: Some(Ident { value: "pg_catalog", quote_style: None }), name: "+" }), right: Value(Number("4")) })], from: [], lateral_views: [], selection: None, group_by: [], having: None }), order_by: [], limit: None, offset: None, fetch: None })'
+- input: select 2 < 3 operator(+) 4;
+  formatted_sql: SELECT 2 < 3 OPERATOR(+) 4
+  formatted_ast: 'Query(Query { with: None, body: Select(Select { distinct: All, projection: [UnnamedExpr(BinaryOp { left: Value(Number("2")), op: Lt, right: BinaryOp { left: Value(Number("3")), op: PGQualified(QualifiedOperator { schema: None, name: "+" }), right: Value(Number("4")) } })], from: [], lateral_views: [], selection: None, group_by: [], having: None }), order_by: [], limit: None, offset: None, fetch: None })'
+- input: select 2 * 3 operator;
+  formatted_sql: SELECT 2 * 3 AS operator
+  formatted_ast: 'Query(Query { with: None, body: Select(Select { distinct: All, projection: [ExprWithAlias { expr: BinaryOp { left: Value(Number("2")), op: Multiply, right: Value(Number("3")) }, alias: Ident { value: "operator", quote_style: None } }], from: [], lateral_views: [], selection: None, group_by: [], having: None }), order_by: [], limit: None, offset: None, fetch: None })'
+- input: select 2 < 3 operator;
+  formatted_sql: SELECT 2 < 3 AS operator
+  formatted_ast: 'Query(Query { with: None, body: Select(Select { distinct: All, projection: [ExprWithAlias { expr: BinaryOp { left: Value(Number("2")), op: Lt, right: Value(Number("3")) }, alias: Ident { value: "operator", quote_style: None } }], from: [], lateral_views: [], selection: None, group_by: [], having: None }), order_by: [], limit: None, offset: None, fetch: None })'
```

Comment:
PostgreSQL actually allows the former but rejects the latter, while we always peek for `(` after `operator`. The relationship between `get_next_precedence` and `parse_infix` within `sqlparser` makes it hard to replicate this behavior.